app/src/main/java/unipd/se18/ocrcamera/ZXingBarcode.java

```diff
@@ -0,0 +1,80 @@
+package unipd.se18.barcodemodule;
+
+import com.google.zxing.BinaryBitmap;
+import com.google.zxing.ChecksumException;
+import com.google.zxing.FormatException;
+import com.google.zxing.common.HybridBinarizer;
+import com.google.zxing.LuminanceSource;
+import com.google.zxing.MultiFormatReader;
+import com.google.zxing.NotFoundException;
+import com.google.zxing.Reader;
+import com.google.zxing.Result;
+import com.google.zxing.RGBLuminanceSource;
+import android.graphics.Bitmap;
+import android.util.Log;
+
+/**
+ * This class implements the ZXing API for barcode recognition
+ * More at https://github.com/zxing/zxing
+ * APIDocs at https://zxing.github.io/zxing/apidocs/overview-summary.html
+ * @author Elia Bedin
+ */
+
+public class ZXingBarcode implements Barcode {
+
+    //The string decoded from the barcode.
+    private String code = "";
+    //Text of the decoding error
+    private static final String DECODE_ERROR =
+            "ERROR: Barcode decoding unsuccessful, please try again.";
+
+
+    /**
+     * implementation of the decodeBarcode method, that would detect the barcode from the given
+     * image
+     * @param bitmap photo taken from the camera, to be analyzed.
+     */
+    @Override
+    public String decodeBarcode(Bitmap bitmap) {
+
+        //int array needed by the RGBLuminanceSource constructor
+        int[] pixels = new int[bitmap.getWidth()*bitmap.getHeight()];
+        //getPixels copies pixel data from the Bitmap into the 'pixels' int array
+        //The three zeros (that will always be zeros) are, in order:
+        //  offset, the first index to write into pixels[];
+        //  the x coordinate of the first pixel to read from the bitmap;
+        //  the y coordinate of the first pixel to read from the bitmap
+        bitmap.getPixels(pixels, 0, bitmap.getWidth(), 0, 0, bitmap.getWidth(),
+                bitmap.getHeight());
```

Comment:
Event if the three zeros won't change their value in the future if I were you, I've put them into three constants each one with the corresponding name you wrote in the comment. The code is stronger and if, for some reason the comment will be erased, who reads the code can immediately understand what is going on.

Reply:
I'm asking you this because you definitely have more experience than me coding.
I didn't wrote this static zeros in constants because I thought they would just make the variable declaration cluttered, since this zeros have really no meaningful information.
If, however, you say that it is better practice anyway, even for this type of costants, I'll change it.